warehouse/oidc/forms.py

```diff
@@ -22,8 +22,12 @@
 _VALID_GITHUB_REPO = re.compile(r"^[a-zA-Z0-9-_.]+$")
 _VALID_GITHUB_OWNER = re.compile(r"^[a-zA-Z0-9][a-zA-Z0-9-]*$")
 
+_VALID_PROJECT_NAME = re.compile(
+    r"^([A-Z0-9]|[A-Z0-9][A-Z0-9._-]*[A-Z0-9])$", re.IGNORECASE
+)
```

Comment:
I feel like maybe this is already defined somewhere else and should come from there 😉

Reply:
Yeah, definitely...I'll find and reuse it 😅

Reply:
Looks like there wasn't actually a shared one! I've deduped a few of them and put it under `warehouse.utils.project.PROJECT_NAME_RE`.

Reply:
b52f3e71d